### PR TITLE
[hash-chain] Add types

### PR DESCRIPTION
**Pull request by BendingBender**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/hash-chain/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+ * fs.write(fd, chain.offset, 0, 4)
+ * fs.write(fd, chain.chain)
+ * fs.close(fd)
+ *
+ * const data = fs.readFile('filename')
+ * const chain = new HashChain(data.subarray(4), data.readUInt32LE(0))
+ */
+declare class HashChain implements Iterable<Buffer> {
+    static readonly SEEDBYTES: number;
+    static readonly BYTES: number;
```
Comment (peterblazejewicz) [inline types/hash-chain/index.d.ts:29]:
```suggestion
    static readonly SEEDBYTES: 32;
    static readonly BYTES: 32;
```
we can be explicit (optionally)

File: types/hash-chain/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "target": "ES2015",
```
Comment (peterblazejewicz) [inline types/hash-chain/tsconfig.json:7]:
```suggestion
        "target": "es6",
```
same, probably the reason of 'check config' tag

Comment (BendingBender) [reply types/hash-chain/tsconfig.json]: done

File: types/hash-chain/index.d.ts (hunk 1 above)
Comment (BendingBender) [reply types/hash-chain/index.d.ts]: done

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by BendingBender** [comment]
Ready to merge

**Pull request merged**